.github/workflows/libfuzzer.yaml

```diff
@@ -0,0 +1,170 @@
+      with:
+        name: config.log for PostgreSQL
+        path: ~/${{ env.PG_SRC_DIR }}/config.log
+
+    - name: Build TimescaleDB
+      run: |
+        set -e
+
+        export LIBFUZZER_PATH=$(dirname "$(readlink -e "$(readlink -e $(llvm-config --libdir)/clang/*/lib/linux/libclang_rt.fuzzer_no_main-x86_64.a | head -1)")")
+
+        cmake -B build -S . -DASSERTIONS=ON -DLINTER=OFF -DCMAKE_VERBOSE_MAKEFILE=1 \
+            -DWARNINGS_AS_ERRORS=1 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_C_COMPILER=clang \
+            -DCMAKE_C_FLAGS="-fsanitize=fuzzer-no-link -lstdc++ -L$LIBFUZZER_PATH -l:libclang_rt.fuzzer_no_main-x86_64.a -static-libsan" \
+            -DPG_PATH=$HOME/$PG_INSTALL_DIR
+
+        make -C build -j$(nproc) install
+
+    - name: Run libfuzzer for compression
+      run: |
+        set -xeu
+
+        mkdir db
+        export PGDATA=db
+        export PGPORT=5432
+        export PGDATABASE=postgres
+        export PATH=$HOME/$PG_INSTALL_DIR/bin:$PATH
+        initdb
+        echo "shared_preload_libraries = 'timescaledb'" >> $PGDATA/postgresql.conf
+        pg_ctl -l postmaster.log start
+
+        psql -c "create extension timescaledb;"
+
+        # Create the fuzzing function
+        export MODULE_NAME=$(basename $(find $HOME/$PG_INSTALL_DIR -name "timescaledb-tsl-*.so"))
+        psql -a -c "create or replace function fuzz(algo cstring, type regtype, runs int) returns int as '"$MODULE_NAME"', 'ts_fuzz_compression' language c;"
+
+        # Symlink the corpus directory to the database directory.
+        ln -sf $(readlink -e tsl/test/fuzzing/compression/gorilla-float8) $PGDATA/corpus
+
+        # The LLVM fuzzing driver calls exit(), so we expect to lose the connection.
+        ret=0
+        psql -v ON_ERROR_STOP=1 -c "select fuzz('gorilla', 'float8', 100000);" || ret=$?
+        if ! [ $ret -eq 2 ]
+        then
+            >&2 echo "Unexpected psql exit code $ret"
+            exit 1
+        fi
+
+        # Check that the server is still alive.
+        psql -c "select 1"
+
+
+    - name: Collect the logs
+      if: always()
+      id: collectlogs
+      run: |
+        find . -name postmaster.log -exec cat {} + > postgres.log
+        # wait in case there are in-progress coredumps
+        sleep 10
+        if coredumpctl -q list >/dev/null; then echo "coredumps=true" >>$GITHUB_OUTPUT; fi
+        # print OOM killer information
+        sudo journalctl --system -q --facility=kern --grep "Killed process" || true
+
+    - name: Save PostgreSQL log
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: PostgreSQL log
+        path: postgres.log
+
+    - name: Save fuzzer-generated cases
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: Fuzzer-generated cases
+        path: db/crash-*
+
+    - name: Stack trace
+      if: always() && steps.collectlogs.outputs.coredumps == 'true'
+      run: |
+        sudo coredumpctl gdb <<<"
```

Comment:
It is probably beyond the scope of this PR. But it might be good to move the gdb instructions to a script and unify the handling of the coredumps. It seems we already have two different ways of doing it. This approach might only analyze one coredump, while [this approach](https://github.com/timescale/timescaledb/blob/2bd18b8e44e6bc6d2ef57aebc2a3e22ba91059b3/.github/workflows/linux-32bit-build-and-test.yaml#L120) loops over all present core files.

Reply:
We'll need a different script for sanitizers because there are other interesting stack frames between ExceptionalCondition. For this one as well, ubsan is enabled by default when enabling fuzzer, so we might want to add the search for sanitizer frames here.

Reply:
Just an idea: We could have a generic script that iterates over all core dumps and performs some generic analysis of the stack frames / variables. Based on an optional parameter, it performs additional gdb commands like the mentioned analysis for additional stack frames. 

This way we would have all the core dump-related actions in one script and not spread them all over the actions.

Reply:
Agree with @jnidzwetzki but this can be part of a following PR right? We'll have some fun already enabling this CI